x-pack/plugins/security_solution/server/endpoint/routes/metadata/support/unenroll.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller, IScopedClusterClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { metadataMirrorIndexPattern } from '../../../../../common/endpoint/constants';
+import { EndpointStatus } from '../../../../../common/endpoint/types';
+
+const KEEPALIVE = '30s';
+const SIZE = 1000;
+
+export interface HostId {
+  host: {
+    id: string;
+  };
+}
+
+interface HitSource {
+  _source: HostId;
+}
+
+export async function findUnenrolledHostByHostId(
+  client: IScopedClusterClient,
+  hostId: string
+): Promise<HostId | undefined> {
+  const queryParams = {
+    index: metadataMirrorIndexPattern,
+    body: {
+      size: 1,
+      _source: ['host.id'],
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                'Endpoint.status': EndpointStatus.unenrolled,
+              },
+            },
+            {
+              term: {
+                'host.id': hostId,
+              },
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  const response = (await client.callAsCurrentUser('search', queryParams)) as SearchResponse<
+    HostId
+  >;
+  const newHits = response.hits?.hits || [];
+
+  if (newHits.length > 0) {
+    const hostIds = newHits.map((hitSource: HitSource) => hitSource._source);
+    return hostIds[0];
+  } else {
+    return undefined;
+  }
+}
+
+export async function findAllUnenrolledHostIds(client: IScopedClusterClient): Promise<HostId[]> {
+  const queryParams = {
+    index: metadataMirrorIndexPattern,
+    scroll: KEEPALIVE,
+    body: {
+      size: SIZE,
+      _source: ['host.id'],
+      query: {
+        bool: {
+          filter: {
+            term: {
+              'Endpoint.status': EndpointStatus.unenrolled,
+            },
+          },
+        },
+      },
+    },
+  };
+  const response = (await client.callAsCurrentUser('search', queryParams)) as SearchResponse<
+    HostId
+  >;
+
+  // eslint-disable-next-line no-return-await
+  return await fetchAllUnenrolledHostIdsWithScroll(response, client.callAsCurrentUser);
```

Comment:
nit: I think you can remove the `await` since the caller of this function is doing an `await` right? Unless we're doing this for call stack debugging?